syntax = "proto3";

package csm.consensus;

message RawRequestVote
{
    uint64 term = 1;    // 候选人的任期
    uint32 candidateId = 2;     // 发起投票的候选人的ID
    uint64 lastLogIndex = 3;        // 候选者最后一个日志条目的索引
    uint64 lastLogTerm = 4;     // 候选者最后一个日至条目是在哪个任期产生的
}

message RawRequestVoteReply
{
    uint64 term = 1;    // 本服务器当前任期，供请求发起者自行决断是否更新
    bool voteGranted = 2;   // true表示候选人获得选票，即我愿意将票投给你
}